src/main/java/frc/robot/hardware/request/cansparkmax/SparkDoubleRequest.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.hardware.request.cansparkmax;
+
+import com.revrobotics.CANSparkBase;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.hardware.request.IRequest;
+
+import java.util.function.BiFunction;
+
+public class SparkDoubleRequest implements IRequest<Double> {
+
+	public enum SparkDoubleRequestType {
+
+		CURRENT(CANSparkBase.ControlType.kCurrent),
+		VOLTAGE(CANSparkBase.ControlType.kVoltage),
+		POSITION(CANSparkBase.ControlType.kPosition);
```

Comment:
why position? the motor always return rotations

Reply:
u need a no rotation2d pos, like elevator pos

Reply:
motor and elevator are different things, this class only talking to motor that returns position as rotations